### PR TITLE
Proxify geoserver

### PR DESCRIPTION
**Pull request by danduk82**
be able to use any URL as baseUrl for geoserver, it is useful if there are other proxies in front, or if the ingress is already used by other routes and you have to add geoserver cloud "nicely" and not as the only service served by the ingress.

### CODE REVIEW

File: templates/_helpers.tpl [hunk 1]
```diff
@@ -68,7 +68,7 @@ Create the name of the service account to use
 Generate content of spring config for rabbitmq
 */}}
 {{- define "geoserver.spring.rabbitmq" -}}
-host: {{ include "geoserver.fullname" . }}-rabbitmq
+host: {{ .Release.Name }}-rabbitmq
```
Comment (sbrunner) [inline templates/_helpers.tpl:71]: With that, you can't include tow times the chart in another chart with alias, but it's probably not the goal :-)

File: templates/config/configmap.yaml [hunk 2]
```diff
@@ -207,88 +207,113 @@ data:
...
+          - StripBasePath=${geoserver.base-path} #remove the base path on downstream requests
           actuator:
             verbose:
               enabled: true
           routes:
           - id: catalog
             uri: lb://catalog-service
             predicates:
-            - Path=/api/v1/**
+            - Path=${geoserver.base-path}/api/v1/**
```
Comment (sbrunner) [inline templates/config/configmap.yaml:249]: Should be to be interpreted by Helm, then I suppose that's interpreted by Geoserver :-)
```suggestion
            - Path={{geoserver.base-path}}/api/v1/**
```

**Review by sbrunner — APPROVED** [review]
Just some notice :-)

File: templates/config/configmap.yaml (hunk 2 above)
Comment (danduk82) [reply templates/config/configmap.yaml:249]: yes, it's actually used by spring boot

File: templates/_helpers.tpl (hunk 1 above)
Comment (danduk82) [reply templates/_helpers.tpl:71]: in the end we should have a dedicated rabbitmq imo

**Pull request merged**